tests/foreman/endtoend/test_cli_endtoend.py

```diff
@@ -0,0 +1,406 @@
+            u'id': activation_key['id'],
+            u'organization-id': org['id'],
+            u'value': '1',
+        })
+
+        # BONUS: Create a content host and associate it with promoted
+        # content view and last lifecycle where it exists
+        content_host = self._create(
+            user,
+            ContentHost,
+            {
+                u'content-view-id': content_view['id'],
+                u'lifecycle-environment-id': lifecycle_environment['id'],
+                u'name': gen_alphanumeric(),
+                u'organization-id': org['id'],
+            }
+        )
+        # check that content view matches what we passed
+        self.assertEqual(
+            content_host['content-view'],
+            content_view['name']
+        )
+        # check that lifecycle environment matches
+        self.assertEqual(
+            content_host['lifecycle-environment'],
+            lifecycle_environment['name']
+        )
+
+        # step 2.15: Create a new libvirt compute resource
+        self._create(
+            user,
+            ComputeResource,
+            {
+                u'name': gen_alphanumeric(),
+                u'provider': u'Libvirt',
+                u'url': u'qemu+tcp://{0}:16509/system'.format(
+                    settings.server.hostname),
+            }
+        )
+
+        # step 2.16: Create a new subnet
+        subnet = self._create(
+            user,
+            Subnet,
+            {
+                u'name': gen_alphanumeric(),
+                u'network': gen_ipaddr(ip3=True),
+                u'mask': u'255.255.255.0',
+            }
+        )
+
+        # step 2.17: Create a new domain
+        domain = self._create(
+            user,
+            Domain,
+            {u'name': gen_alphanumeric()}
+        )
+
+        # step 2.18: Create a new hostgroup and associate previous entities to
+        # it
+        host_group = self._create(
+            user,
+            HostGroup,
+            {
+                u'domain-id': domain['id'],
+                u'name': gen_alphanumeric(),
+                u'subnet-id': subnet['id'],
+            }
+        )
+        if not bz_bug_is_open('1326101'):
+            Org.with_user(
+                user['login'],
+                user['password']
+            ).add_hostgroup({
+                u'hostgroup-id': host_group['id'],
+                u'id': org['id'],
+            })
+
+        # step 2.19: Provision a client
+        self.client_provisioning(activation_key['name'], org['label'])
+
+    def _create(self, user, entity, attrs):
+        """Creates a Foreman entity and returns it.
+
+        :param dict user: A python dictionary representing a User
+        :param obj entity: A valid CLI entity.
+        :param dict attrs: A python dictionary with attributes to use when
+            creating entity.
+        :return: A ``dict`` representing the Foreman entity.
+        :rtype: dict
+        """
```

Comment:
Actually according to PEP8 there is no need to add a blank line here https://www.python.org/dev/peps/pep-0008/#documentation-strings.